### PR TITLE
scripts,workflows: add fossa wrapper script for config generation and retries

### LINKED ISSUE #4240: FOSSA workflow in Pull Requests is flaky
Looking at the [history of FOSSA workflow runs](https://github.com/backstage/backstage/actions?query=workflow%3AFOSSA), they sometimes fail due to server error.

[Example](https://github.com/backstage/backstage/runs/1760655356?check_suite_focus=true)
```
INFO Uploading analysis...
FATAL Error during upload: Error response from API: <html>
<head><title>502 Bad Gateway</title></head>
<body>
<center><h1>502 Bad Gateway</h1></center>
<hr><center>nginx</center>
</body>
</html>
```

# Possible Solution

1. Few delayed retries when upload fails due to errors.

Previous discussion
https://github.com/backstage/backstage/pull/4077#issuecomment-764463876

### PR DESCRIPTION
**Pull request by Rugvip**
Fixes #4240 😅

### CODE REVIEW

**Comment by backstage-service** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr4723-7xwnrfpcuvr2xyuyn6pn3gtwsnwuznbf.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/603a461b65f8175d96fc9015

**Comment by backstage-service** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr4723-7xwnrfpcuvr2xyuyn6pn3gtwsnwuznbf.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/603a461b65f8175d96fc9015

**Comment by backstage-service** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr4723-7xwnrfpcuvr2xyuyn6pn3gtwsnwuznbf.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/603a461b65f8175d96fc9015

File: scripts/run-fossa.js [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// This script generates an appropriate fossa config, and wraps the running
+// of `fossa analyze` in a retry loop as if frequently fails with a 502 error
```
Comment (freben) [inline scripts/run-fossa.js:19]:
```suggestion
// of `fossa analyze` in a retry loop as it frequently fails with a 502 error
```

File: scripts/run-fossa.js [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+      if (!error.code) {
+        throw error;
+      }
+      if (error.stderr) {
+        process.stderr.write(error.stderr);
+      }
+      if (error.stdout) {
+        process.stdout.write(error.stdout);
+      }
+      if (error.stderr.includes('502 Bad Gateway')) {
```
Comment (freben) [inline scripts/run-fossa.js:104]:
```suggestion
      if (error.stderr?.includes('502 Bad Gateway')) {
```

Or sure to not be needed?

**Review by freben — APPROVED** [review]
Awesome

File: scripts/run-fossa.js (hunk 1 above)
Comment (Rugvip) [reply scripts/run-fossa.js]: Just gotta read it upside down :grin:

File: scripts/run-fossa.js (hunk 2 above)
Comment (Rugvip) [reply scripts/run-fossa.js]: Yep check is needed, optional chaining is only available from node 14 tho :<

**Comment by backstage-service** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr4723-7xwnrfpcuvr2xyuyn6pn3gtwsnwuznbf.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/603a461b65f8175d96fc9015

**Pull request merged**